### PR TITLE
Support nesting for `catch-error-name` rule

### LINKED ISSUE #107: `catch-error-name` fails when try/catch is inside a promise that is caught
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/107)

Discovered in https://github.com/sindresorhus/got/pull/347

Failing tests: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/9f9a324255cc1faaa0d5331a6b78be0cbec9db8f

The cause is the `.catch(err => err);` part. If that is removed it correctly errors that `e` should be named `err`.

// @SamVerschueren

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Submitted pull Requests
- [#227 Support nesting for catch-error-name rule](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/227)
- [#653 `catch-error-name`: Add nesting tests](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/653)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by doniyor2109**
Resolves #107

### CODE REVIEW

**Comment by sindresorhus** [comment]
Since almost nobody actually volunteered to review other PRs, I've decided to make it a requirement. You have to review at least two other PRs to get your own PR merged. Thank you for understanding.

https://github.com/sindresorhus/eslint-plugin-unicorn/pull/226#issuecomment-459012736

File: rules/catch-error-name.js [hunk 1]
```diff
@@ -47,18 +36,36 @@ const create = context => {
 	const stack = [];
 
 	function push(value) {
-		if (stack.length === 1) {
-			stack[0] = true;
+		stack.push(value);
+	}
+
+	function indexifyName(name, scope) {
```
Comment (sindresorhus) [inline rules/catch-error-name.js]: Would be better to reuse https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/utils/avoid-capture.js here instead. See: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/257 // @futpib

Comment (sindresorhus) [reply rules/catch-error-name.js]: Ping @doniyor2109

Comment (MrHen) [reply rules/catch-error-name.js]: This was done as part of https://github.com/sindresorhus/eslint-plugin-unicorn/pull/306 which is almost certainly what is causing the merge conflict.

**Pull request closed without merging**